### ISSUE TITLE
Can freefalcon-central be built with MSVS2013 and directx9 or directx10?

### ISSUE THREAD

**Issue by kinsung** [FreeFalcon/freefalcon-central#45]
Can freefalcon-central be built with MSVS2013 and directx9 or directx10?

**Comment by artscout** [comment]
Just for fun and as learning experience I;m working on fixing it for VS2026 and DX11. I will post PR when I'm done, Right now build is working, fixing textures/lights as ground textures and lights not working currently along with particles system.

<img width="1018" height="797" alt="Image" src="https://github.com/user-attachments/assets/67f2d2f8-5c8a-4d72-a87d-2b8e3a5b9e68" /> WIP screenshot (light system WIP so everything is black, MFD WIP, DED/RWR WIP)

**Comment by artscout** [comment]
JFYI I used to be part fo the team back in 2009, however briefly so I do know it somewhat, but well BMS is the best instance anyway so this is just fun experiment for me. Also well if we reanimate is somewhat mnight be fun for some hardcore funs of FF ) Sadly I lost my archives with hitiles, Aeyes cockpit, SRTM and everything so I don't even remember whats needed for the best picture ) Will see after work is done

**Comment by gordilloedwin** [comment]
hey @artscout  after many years I became somewhat proficient in VS and C++ (curently an Staff Engineer, mostly C#) but if you are willing to do some work, I can offer help... just for the fun of it

I think I've got a ton of files still saved somewhere

**Comment by artscout** [comment]
> hey [@artscout](https://github.com/artscout) after many years I became somewhat proficient in VS and C++ (curently an Staff Engineer, mostly C#) but if you are willing to do some work, I can offer help... just for the fun of it
> 
> I think I've got a ton of files still saved somewhere

Hi there! Well the thing is that I jsut plan to make it running on DX11 and maybe add OpenXR just for fun and just to se if I could. But  I admit that I;m not that good coder really, as I;m system administator and atm mostly write in pythoin and develop world of warcraft plugins in LUA ))) But I managed to get it running with DX11 now and doing the bug fixing right now (it's so full of bugs it's not even funny). I will commit my updates in my fork soon so feel free to join up just for fun of it. We even might create fun release but I feel that we're so far behind BMS there's no point in it except jus to satisfy old hardcore FF enjoyers if there are any.

**Comment by gordilloedwin** [comment]
I'd be willing to put some hours into bringing the old FF alive, of course, I know it's a road to nowhere, but it seems like it's fun...

**Comment by artscout** [comment]
Well this is showcase video if it as it is right now: https://youtu.be/FW1UoN-O5ko ( I run this from debugger just to capture showscase)

**Comment by artscout** [comment]
Also I've decided to use AI to search for bugs (claude) as I have subscription anyway) so we might get it better )

**Comment by gordilloedwin** [comment]
The video on youtube !! is awesome !!! I loved it!! 

yeah, I think AI can help a lot... did you create a branch ? you should upload it...

**Comment by artscout** [comment]
Yeah I did and I will upload it soon. Right after AI will go and fix bugs that were eitherexisting or I introduced them with port. Probably on weekend

**Comment by Stewie23** [comment]
Heya I recently tried to get it to build with codex help. Eager to see what you come up with

**Comment by artscout** [comment]
I hope that you guys won't be disappointed ) Because I really worked hard on it and now just fixing bugs with AI help ) Also well there are some improvements. 
And BTW if any of old guys from FF team reading this - hello from Arkstone :)

**Comment by Stewie23** [comment]
I would be interested in working on externalising the data, as of now it's hard coded . So looking forward to your repo.

**Comment by artscout** [comment]
Ok guys I've pushed dx11port branch on my forked repo. ALso I'll make small patch msi with compiled exe and new data files because I made few changes

**Comment by artscout** [comment]
Keep in mind that while this is starts and runs (at least tactical engagements and instant action do) it may have bugs and CTD's. Well I've tried to fly one mission in campaign and managed to do it but still it's just an experiment so yeah, I just made an update, nothing more

**Comment by artscout** [comment]
Ok guys, I've put update with X64 support and bunch of bugfixes.

**Comment by artscout** [comment]
final update will be soon. there will be no more x86, only x64. there will be OpenXR + QuadViews support (I did it but now I;m working on synchronizing positions, this is proving troubling so I will use claude again to help me with it). Also I will ask claude to make small application to convert voice sounds from st80 and it will be included in code. (it's easy enough so I'll just use claude code generator, much faster this way)

**Comment by Stewie23** [comment]
Awesome, im try to get into contact with the peeps that where able to merge and see to it to get the rights to merge your stuff.

**Comment by artscout** [comment]
Yeah sure. Meanwhile I've got it working even with QuadViews support but still fighting with the mouse of all things. I'll got read another tuttorial maybe I'll find how to do it right ) DCS and BMS did somehow so this is not impossilbe )

**Comment by gordilloedwin** [comment]
@artscout ... hey, I did a quick code review in your branch and so far it looks excellent, very nice job!... I was wondering if I could do maybe a dumb request, to ensure even single statements on code, to add brackets, make it easier for humans to read... that is, if you have the time, of course... and add the line breaks too ... like there is an if like 

```c
if (vc) // JB 010113
            _stprintf(flight_ptr->aircraft_name, vc->Name);
            { _sntprintf(flight_ptr->aircraft_name, sizeof(flight_ptr->aircraft_name)-1, "%s", vc->Name); flight_ptr->aircraft_name[sizeof(flight_ptr->aircraft_name)-1]=0; }
```

is that possible? you can always say no :)

**Comment by artscout** [comment]
It's the whole code formatted this way unfortunately. Might have to ask AI to do it )

**Comment by Stewie23** [comment]
There  where some formatting guids,  check if the suggestion above is in therey and thenshould probably feed them to the ai :)

https://github.com/FreeFalcon/docs/blob/master/standards/syntax.md

**Comment by artscout** [comment]
AI sometimes is pretty dumb about this basic things and can break something out. With this big codebase it's a real possibility. Also it "forgets" formatting tasks when "fixing something" I ask him to find the source of. But it's really usedful for bug hunting and for improvements over base I did. I'd probably still be working on QuadViews without it because this is really big topic. So I did what I could for now but formatting will wait probably as it better to do by hand. I;ve also put release patch on my repo for x64 + openxr and other things like setup pages. I also put resources that I've changed into repo installer folder

**Comment by artscout** [comment]
I also pretty much done with it, maybe some small fixes and rearrangements, aybe will do something with LOD and texture load but that's it for now. FF is still mostly dead and as I said before it's so far behind BMS that there's no point to do anything, also we do not have permission from MicroProse (unlike BMS team) so they can pretty much close this project if they don't like something (I guess they do not care now but that can change at any moment) so we can jsut have some fun with it as long as we can but it's on shaky ground legally so anything can happen and it can be taken down.

**Comment by artscout** [comment]
For now it should work just fine and relatively CTD free unles I missed something. Original was pretty much CTD on CTD because well we lacked some instruments that are available right now (like Application Verifier for example, it helped to find just some UGLY bugs that we couldn't find back then), never mind AI )

**Comment by Stewie23** [comment]
It's still a nice playground :)

But yeah we probably won't get the things integrated in this repo, no luck with the mails. If I get around externalising data I will make a pull request to your fork

**Comment by artscout** [comment]
Sure. Well I found more bugs and I shall work on them this week (already do but it is slow, at least heat wave in Europe passed and my pc isn’t on the verge of overheating, I literally had 28 degrees centigrade at home with my 5090 and pimax crystal super). As they are terrain based I have plans to look at what can be improved there. And few experiments to see what I can do with it graphics wise

**Comment by vinnydiehl** [comment]
@Stewie23 @artscout Sorry, just seeing this! I'll give you both write access to these repos.

**Comment by artscout** [comment]
I've made a new small video using OpernXR plugin for OBS, don't mind my bad english, it;'s small showcase again: https://www.youtube.com/watch?v=o5CEAH5dBy4

**Comment by artscout** [comment]
Also DX11 was the middle step before DX12, as I need DX12 for OpenXR to use single-step rendering instead of rendering 2 or 4 views (stereo or quadviews - each eye requires rendering of the whole screen for eye and gaze zone). With DX12 this gets much better, but it still has its own can of worms.

**Comment by artscout** [comment]
Anyway, does anyone has any idea of what to use to edit LOD? Need to fix cockpit, IFF panel but with what - LOD editor 6.23 doesn't want to load it.

**Comment by artscout** [comment]
My commit and merge closed this issue, I've decided to keep it open for a time being.
I've merged VS 2026 and DX 11/12 code. By default it runs with DirectX12, DX11 used for intermediate path for DX12 and I will clear it later. 
The game itself runs ok, so far I haven't had any CTD but I haven't really played capaigns yet. There might be visual bugs in interface, will address that later.
Known issues:
1. Terrain height pops - this is issue that was in original code, I;ve tried to fix that but my fix makes ground shimmer and I made middle-ground fix for now. Also in some places there's once again height drops for no reason with objects hanging in the air (like bridges or buildings) - I will adress that too. 
2. Runway textures might be overlapped with underlaying ground textures on some angles - this is new bug from DX12 migration, will fix that later. for now we can use flags:  **set g_fGpuTerrainSlopeBias ** and **set g_fGpuTerrainDepthBias** if needed (you can play with values, default is 4 for slopebias and 0 for depth) in ffviper.cfg and play with values till issue disappears. WIP
3. No chatter - need to compile tools\st80conv and run this exe on file sounds'falcon.tlk to convert it to PCM. Needs st80.dll from FF6 install too.
4. There's my work on Touch controllers and new effects in progress, right now only "VR hands" are working correctly so they set as default. New explosions are there for Aircraft and ground but they are WIP and mixed with the old ones. Ground one might look weird, I'll fix that later

**Comment by artscout** [comment]
Ah small comment - D12 is not default yet. And d11 is partially broken ATM - add "set g_bUseD3D12 1" to ffviper.cfg. I;m cleaning out d11 now.

**Comment by artscout** [comment]
Ok I've cleaned up DX11 as it is not needed anymore. Also fixed small bug with menu in VR.

**Comment by artscout** [comment]
And my latest video https://www.youtube.com/watch?v=-PrH7QsTzWU

**Comment by artscout** [comment]
Ok guys this is where I leave it as I've done what I wanted and what I could. Feel free to ask questions and such but as it is I will be just experimenting on stuff locally and not doing anything else.

**Comment by artscout** [comment]
ok I've pushed latest fixed and experiments here. So now it has dx12/vulkan and linux support with OpenXR. and Mesh shaders for terrain. Have fun! )
